### PR TITLE
Base price storage website price mode

### LINKED ISSUE #30132: [Issue] Base price storage website price mode
This issue is automatically created based on existing pull request: magento/magento2#30126: Base price storage website price mode

---------

Fix V1/products/base-prices endpoint to update prices for Catalog Price Mode - Website

### Description (*)
V1/products/base-prices endpoint does not work with Catalog Price Mode - Website
Function was added to affect prices for all store views inside changed website

### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by wassuupp**
Fix V1/products/base-prices endpoint to update prices for Catalog Price Mode - Website

### Description (*)
V1/products/base-prices endpoint does not work with Catalog Price Mode - Website
Function was added to affect prices for all store views inside changed website

### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#30132: Base price storage website price mode

### CODE REVIEW

**Comment by sidolov** [comment]
@magento create issue

File: app/code/Magento/Catalog/Model/Product/Price/BasePriceStorage.php [hunk 1]
```diff
@@ -225,4 +262,32 @@ private function retrieveValidPrices(array $prices)
...
     }
+
+    /**
+     * If Catalog Price Mode is Website, price needs to be applied to all Store Views in this website.
+     *
+     * @param array $formattedPrices
+     * @return array
+     * @throws NoSuchEntityException
+     */
+    private function applyWebsitePrices($formattedPrices)
```
Comment (BarnyShergold) [inline app/code/Magento/Catalog/Model/Product/Price/BasePriceStorage.php]: Missing return type

**Comment by BarnyShergold** [comment]
@magento run all tests

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Failing tests need to be addressed and fixed

File: app/code/Magento/Catalog/Model/Product/Price/BasePriceStorage.php (hunk 1 above)
Comment (wassuupp) [reply app/code/Magento/Catalog/Model/Product/Price/BasePriceStorage.php]: added

**Comment by wassuupp** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@wassuupp could you please fix failed Static and Unit tests?
Thank you.

**Comment by engcom-Charlie** [comment]
@wassuupp please fix failed Static and Unit tests? Otherwise, we can't proceed with your PR.
Thank you.

**Comment by sidolov** [comment]
@wassuupp  I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Comment by nuzil** [comment]
I will proceed with this one

**Comment by nuzil** [comment]
@magento run all tests

**Comment by wassuupp** [comment]
@magento run all tests

**Comment by BarnyShergold** [comment]
@magento run functional tests b2b

**Comment by BarnyShergold** [comment]
@magento run Functional Tests B2B

**Comment by BarnyShergold** [comment]
@wassuupp - The tests keep failing - please can you check these and see if there is any reason why you changes maybe affecting this

**Comment by wassuupp** [comment]
@magento run all tests

**Comment by wassuupp** [comment]
@magento run Unit Tests

**Comment by wassuupp** [comment]
@magento run all tests

**Comment by wassuupp** [comment]
@magento run all tests

**Comment by nuzil** [comment]
Closing as there are full mess in source branch. New one is coming

**Pull request closed without merging**